### ISSUE TITLE
Cleanup handling of potential variables.

### ISSUE THREAD

**Issue by BeksOmega** [google/blockly#5273]
**Is your feature request related to a problem? Please describe.**

Potential variables are variables that we don't want to exist on the main workspace. For example, if you want blocks in your flyout to reference variables, but you don't want those variables to be accessible in the main workspace until the block is dragged into the main workspace.

The problem is, we shouldn't need a separate [potentialVariableMap_](https://github.com/google/blockly/blob/d4196c1105d3d7719e967edb0cecd0d7f76508c0/core/workspace.js#L122) which is only available on workspaces that are flyouts. We should be able to use the normal [variableMap_](https://github.com/google/blockly/blob/d4196c1105d3d7719e967edb0cecd0d7f76508c0/core/workspace.js#L110) of the workspace that is a flyout.

**Describe the solution you'd like**

Use the normal variableMap_ of the workspace that is the flyout, instead of creating a separate property that is only used by flyouts. This would require modifying the [flyout](https://github.com/google/blockly/blob/master/core/flyout_base.js), [workspace](https://github.com/google/blockly/blob/master/core/workspace.js), and [variables namespace](https://github.com/google/blockly/blob/master/core/variables.js).

**Additional context**

The [potential variable map is package](https://github.com/google/blockly/blob/d4196c1105d3d7719e967edb0cecd0d7f76508c0/core/workspace.js#L735), so this shouldn't be a breaking change.

**Comment by maribethb** [comment]
Hey Beka (and team),

after looking into a PR that tries to refactor this, I'm not sure it actually simplifies things. More details here: https://github.com/google/blockly/pull/5656#discussion_r756390362

with the tldr: "It simplifies things in one respect, because we are eliminating the potentialVariableMap property which is unused for main workspaces. However, now we are using one property to mean two different things depending on whether it's a flyout workspace or not, which is a different kind of complexity that comes through on the comment for that second if statement." (the complexity being you still need to check the target workspace's variablemap first)

So I'm leaning towards not changing anything here after all, what do you think?

**Comment by BeksOmega** [comment]
@maribethb yeah I feel you. Having complexity hidden in comments is not a great look. However, having properties of a superclass that is only used by a (sort-of) subclass doesn't smell good either. Imo the property feels like an acretion/band-aid rather than a principled abstraction. Like, the workspace has knowledge of this specific use case the flyout has, which it shouldn't know about, which makes the design feel inelegant.

So I still don't think that the potentialVariableMap is a good solution, and I think I still personally lean toward unifying the properties. But if you think two properties is less confusing than one I totally hear you (explicit complexity is better than implicit!), and that sounds good to me.

**Comment by maribethb** [comment]
Yeah, I hear you, it does feel weird having the workspace know about the potential variable map. After looking at it again, it doesn't seem like it needs to. The flyout could have its own `potentialVariableMap` stored on the flyout class instead of the workspace class, right? Would that be better? I don't think that would significantly change any of the logic when we're checking for variables, but it would move the property from workspace to flyout.

**Comment by BeksOmega** [comment]
Yesss I love that idea! But sadly I don't think it'll work b/c typing :/ I checked and apparently Flyout's aren't actually Workspace subtypes (they just seem to act like it). They are actually [composed with a workspace](https://github.com/google/blockly/blob/e5c40e208075f02f1ffbade8390c4cbc0bb003a7/core/flyout_base.js#L66). And sadly the [variable functions expect workspaces](https://github.com/google/blockly/pull/5656/files#diff-93bef0e469cb8c988511e76706cf55fdfcbd1dcf19ec9f844e056717ecafc8a1R463).

I guess the variable functions *could* be changed to use a type-union of either a Flyout or a Workspace (which would probably be a better representation of the function's itnerface) but I imagine that'll require a lot of changes other places in the code base as well.

So yeah :/ again totally up to you. There doesn't seem to be any really good solution here lol. So whatever you think is best is cool w/ me.
